### PR TITLE
[BTAT-7630] - linking up new dereg date

### PR DESCRIPTION
**Pull request by sraspin**
# Pull Request Checklist
* [ ] Branch is rebased with master
* [ ] Added Unit Tests for changed functionality
* [ ] Ran `sbt clean compile coverage test it:test coverageReport`, all tests pass and coverage meets minimum
* [ ] Checked libraries are up to date with `sbt validate`. (If applicable) library updates are done in a separate, single commit
* [ ] Ran [change-vat-acceptance-tests](https://github.com/hmrc/change-vat-acceptance-tests) (see README of repo)
* [ ] Ran [change-vat-performance-tests](https://github.com/hmrc/change-vat-performance-tests) (see README of repo)
* [ ] (If applicable) raised app-config-* PRs
* [ ] (If applicable) ran Wave test for changes to views
* [ ] (If applicable) added Integration Tests

### CODE REVIEW

File: app/models/CheckYourAnswersModel.scala [hunk 1]
```diff
@@ -145,12 +146,19 @@ case class CheckYourAnswersModel(deregistrationReason: Option[DeregistrationReas
...
+  ))
+
   private val deregDateAnswer = deregDate.map(answer => CheckYourAnswersRowModel(
     messages("checkYourAnswers.question.deregistrationDate"),
-    if(answer.yesNo.value)
-      {Html(answer.date.get.longDate())}
-    else
-      {Html(messages("checkYourAnswers.answer.no"))},
+    Html(answer.longDate),
     controllers.routes.ChooseDeregistrationDateController.show().url,
```
Comment (lewishay) [inline app/models/CheckYourAnswersModel.scala]: As part of running my acceptance tests against this branch I’ve noticed that both of the date Change links go to the same screen. I think this one here needs to be updated to the new screen.

File: it/pages/ChooseDeregistrationDateISpec.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ChooseDeregistrationDateISpec extends IntegrationBaseSpec {
 
         given.user.isAuthorised
 
-        DeregisterVatStub.successfulGetAnswer(vrn, ChooseDeregDateAnswerService.key)(Json.toJson(validYesModel))
+        DeregisterVatStub.successfulGetAnswer(vrn, ChooseDeregDateAnswerService.key)(Json.toJson(Yes))
```
Comment (Tasmine10ten) [inline it/pages/ChooseDeregistrationDateISpec.scala]: Think it's safe to remove lines 33-38 (testDay/testMonth etc) of this file (wouldn't let me comment on the lines themselves as no changes made on them)

File: test/controllers/ChooseDeregistrationDateControllerSpec.scala [hunk 3]
```diff
@@ -28,18 +28,20 @@ import play.api.http.Status.INTERNAL_SERVER_ERROR
 import play.api.mvc.AnyContentAsFormUrlEncoded
 import play.api.test.FakeRequest
 import play.api.test.Helpers.{contentType, _}
-import services.mocks.{MockChooseDeregDateAnswerService, MockOutstandingInvoicesService}
+import services.mocks.{MockChooseDeregDateAnswerService, MockOutstandingInvoicesService, MockWipeRedundantDataService}
```
Comment (Tasmine10ten) [inline test/controllers/ChooseDeregistrationDateControllerSpec.scala]: Same for this page (line 49-53 testDay etc) 
Also on line 117 can remove the date part
("the user submits after selecting a 'Yes' option and a date" should)

File: test/views/ChooseDeregistrationDateSpec.scala [hunk 4]
```diff
@@ -122,76 +113,34 @@ class ChooseDeregistrationDateSpec extends ViewBaseSpec {
...
+      YesNoForm.yesNo -> YesNoForm.yes
+    )))
     lazy implicit val document: Document = Jsoup.parse(view.body)
 
     s"have the correct document title" in {
       document.title shouldBe ChooseDeregistrationDateMessages.title
     }
 
-    "have no error heading message being displayed" in {
+    "have an error heading message being displayed" in {
```
Comment (Tasmine10ten) [inline test/views/ChooseDeregistrationDateSpec.scala]: I think this should be "no" instead of "an"

**Pull request merged**